faucet/src/faucet.rs

```diff
@@ -128,41 +159,37 @@ impl Faucet {
                     if lamports > cap {
                         return Err(Error::new(
                             ErrorKind::Other,
-                            format!("request too large; req: {} cap: {}", lamports, cap),
+                            format!(
```

Comment:
In this case what if we returned a transaction with a memo instruction in it containing a "request to large" message" instead?   I foresee support questions like, "why did I get no SOL if the transaction succeeded?", but this might be a nice way to pull new folks down the Solana rabbit hole too.

Reply:
Oh, just the memo, no transfer? Interesting...
Probably not too much of a stretch to update the cli to handle that sanely; I believe it already uses a balance check to confirm a successful airdrop. Just a matter of printing something about the memo transaction before returning the error.
Might need to adjust retries.

Reply:
Added, ready for another look. Wdyt?

Reply:
Cli example:
```
$ solana-faucet --per-request-cap 1
```

```
$ solana airdrop 2
Requesting airdrop of 2 SOL from 127.0.0.1:9901

Signature: 3pXazHtHyGy15Ay1pA38HYqv4CT7vFvnBRH4fXLGGKDXU9zKtax6bST5SytDcN4pdSR5xxUrvGwhL6Gwzdw4ce9c

Balance unchanged
Run `solana confirm -v <SIGNATURE>` for more info
```

```
$ solana confirm -v 3pXazHtHyGy15Ay1pA38HYqv4CT7vFvnBRH4fXLGGKDXU9zKtax6bST5SytDcN4pdSR5xxUrvGwhL6Gwzdw4ce9c`
RPC URL: http://127.0.0.1:8899
Default Signer Path: id.json
Commitment: confirmed

Transaction executed in slot 4054:
  Block Time: 2021-04-05T21:23:14-06:00
  Recent Blockhash: FzngNtHcUQNBuHy98XS8eAKVvdmbiLbbGtHJnxPYGLre
  Signature 0: 3pXazHtHyGy15Ay1pA38HYqv4CT7vFvnBRH4fXLGGKDXU9zKtax6bST5SytDcN4pdSR5xxUrvGwhL6Gwzdw4ce9c
  Account 0: srw- 55U4CmDWVgLmPaL5MCDjCkT96zWz66mLzN7h6o4E6ejK (fee payer)
  Account 1: -r-x MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr
  Instruction 0
    Program:   MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr (1)
    Data: "request too large; req: 2 SOL cap: 1 SOL"
  Status: Ok
    Fee: ◎0.000005
    Account 0 balance: ◎999998.99997 -> ◎999998.999965
    Account 1 balance: ◎0.000000001
  Log Messages:
    Program MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr invoke [1]
    Program log: Memo (len 40): "request too large; req: 2 SOL cap: 1 SOL"
    Program MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr consumed 15830 of 200000 compute units
    Program MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr success

Confirmed
```

Reply:
And...
```
$ solana airdrop 1
Requesting airdrop of 1 SOL from 127.0.0.1:9900

Signature: 24Fo5yYZrvdzV4nf3HLfndXx8BmF5Va1JK7EHxt5uHDPa2LKAK5P9LzQVfjDwkHCb7E7a19GfJtjCbexDcRBobpH
```

Reply:
😻